canvas_exercise/shapes_game/index.js

```diff
@@ -1,36 +1,135 @@
 window.addEventListener("load", function() {
 
   function clear(ctx, width, heigt) {
+    ctx.clearRect(0, 0, width, height);
   }
 
-  function drawRandomShape(ctx, width, height) {
+  function drawTraingle(ctx, width, height, color) {
+    ctx.fillStyle = color;
+    ctx.beginPath();
+    ctx.moveTo(width,height);
+    ctx.lineTo(width+100, height+100);
+    ctx.lineTo(width, height+100);
+    ctx.fill();
+    ctx.closePath();
   }
 
-  function drawGameStartText(ctx, width, height, score) {
+  function drawRandomShape(shapes) {
+    var shape = shapes[Math.floor(Math.random()*shapes.length)];
+    var randWidth = Math.floor(Math.random() * (700 - 100)) + 100;
+    var randHeight = Math.floor(Math.random() * (650 - 100)) + 100;
+    var eKey;
+    switch(shape) {
+      case "redT":
+          clear(ctx, (canvas.width), (canvas.height));
+          drawTraingle(ctx, randWidth, randHeight, "red");
+          eKey = "redT";
+          break;
+      case "redS":
+          clear(ctx, (canvas.width), (canvas.height));
+          drawSquare(ctx, randWidth, randHeight, 100, 100, "red");
+          eKey = "redS";
+          break;
+      case "whiteT":
+          clear(ctx, (canvas.width), (canvas.height));
+          drawTraingle(ctx, randWidth, randHeight, "white");
+          eKey = "whiteT";
+          break;
+      case "whiteS":
+          clear(ctx, (canvas.width), (canvas.height));
+          drawSquare(ctx, randWidth, randHeight, 100, 100, "white");
+          eKey = "whiteS";
+          break;
+      default:
+          clear(ctx, (canvas.width), (canvas.height));
+          drawTraingle(ctx, randWidth, randHeight, "red");
+          eKey = "redT";
+    }
+    return eKey;
   }
 
-  function restartGame(ctx, width, height) {
+  function drawSquare(ctx, startx, starty, width, height, color) {
+    var square = {
```

Comment:
I don't think you need to make the whole object because you don't really save it and use it later.